langchain/load/serializable.py

```diff
@@ -0,0 +1,95 @@
+from abc import ABC
+from typing import Any, Dict, List, Literal, TypedDict
+
+from pydantic import BaseModel
+
+
+class BaseSerialized(TypedDict):
+    lc: int
+    id: List[str]
+
+
+class SerializedConstructor(BaseSerialized):
+    type: Literal["constructor"]
+    kwargs: Dict[str, Any]
+
+
+class SerializedSecret(BaseSerialized):
+    type: Literal["secret"]
+
+
+class SerializedNotImplemented(BaseSerialized):
+    type: Literal["not_implemented"]
+
+
+class Serializable(BaseModel, ABC):
+    @property
+    def lc_namespace(self) -> List[str]:
+        """
+        Return the namespace of the langchain object.
+        eg. ["langchain", "llms", "openai"]
+        """
+        return self.__class__.__module__.split(".")
+
+    @property
+    def lc_secrets(self) -> Dict[str, str]:
+        """
+        Return a map of constructor argument names to secret ids.
+        eg. {"openai_api_key": "OPENAI_API_KEY"}
+        """
+        return dict()
+
+    lc_kwargs: Dict[str, Any] = dict()
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.lc_kwargs = kwargs
+
+    def to_json(self) -> SerializedConstructor:
+        secrets = dict()
+
+        # Merge the lc_secrets from every class in the MRO
+        for cls in self.__class__.mro():
+            # Once we get to Serializable, we're done
+            if cls is Serializable:
```

Comment:
off by 1 error here likely -- Don't think the secret will get masked if Person is the top object

```
class Person(Serializable):
    secret: str

    @property
    def lc_secrets(self) -> Dict[str, str]:
        return {"secret": "SECRET"}
```

Reply:
No, I tested it, eg for LLMChain `print(self.__class__.mro())`

is 

```
[<class 'langchain.chains.llm.LLMChain'>, <class 'langchain.load.serializable.Serializable'>, <class 'langchain.chains.base.Chain'>, <class 'pydantic.main.BaseModel'>, <class 'pydantic.utils.Representation'>, <class 'abc.ABC'>, <class 'object'>]```